muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/dialogs/new-edit-announcement.tsx

```diff
@@ -392,7 +392,11 @@ class NewEditAnnouncement extends SessionStateComponent<
   createOrModifyAnnouncement(closeDialog: () => any) {
     this.setState({ locked: true });
 
-    if (this.props.status.role === "TEACHER") {
+    const isTeacher = this.props.status.roles
```

Comment:
Unnecessary null/undefined check as roles is always empty list by default. Can use includes checking straight